core/modules/replica/IndexJob.cc

```diff
@@ -0,0 +1,538 @@
+                }
+                f << request->responseData().data;
+                f.close();
+            }
+            break;
+        }
+        case FOLDER: {
+            string const filePath =
+                (_destinationPath.empty() ? "." : _destinationPath) + "/" +
+                database() + "_" + to_string(request->chunk()) + ".tsv";
+            ofstream f(filePath);
+            if (not f.good()) {
+                throw runtime_error(
+                        typeName() + "::" + string(__func__) +
+                        "  failed to open/create file: " + filePath);
+            }
+            f << request->responseData().data;
+            f.close();
+            break;
+        }
+        case TABLE: {
+
+            auto const config = controller()->serviceProvider()->config();
+
+            // ATTENTION: all exceptions which may be potentially thrown are
+            // supposed to be intercepted by a caller of the current method
+            // and be used for error reporting.
+
+            // Dump the data into a temporary file from where it would be loaded
+            // into the MySQL table. Note that the file must be readable by
+            // the MySQL service.
+            //
+            // TODO: consider using the named pipe (FIFO)
+
+            string const filePath =
+                config->qservMasterDatabaseTmpDir() + "/" + database() + "_" +
+                to_string(request->chunk()) +
+                (_hasTransactions ? "_p" + to_string(_transactionId) : "");
+
+            ofstream f(filePath);
+            if (not f.good()) {
+                throw runtime_error(
+                        typeName() + "::" + string(__func__) +
+                        "  failed to open/create file: " + filePath);
+            }
+            f << request->responseData().data;
+            f.close();
+
+            // Open the database connection if this is the first batch of data
+            if (nullptr == _conn) {
+                _conn = database::mysql::Connection::open(
+                    database::mysql::ConnectionParams(
+                        config->qservMasterDatabaseHost(),
+                        config->qservMasterDatabasePort(),
+                        "root",
+                        Configuration::qservMasterDatabasePassword(),
+                        "qservMeta"
+                    )
+                );
+            }
+            string const query =
+                "LOAD DATA INFILE " + _conn->sqlValue(filePath) +
+                " INTO TABLE " + _conn->sqlId(_destinationPath);
+
+            _conn->execute([&](decltype(_conn) conn) {
+                conn->begin();
+                conn->execute(query);
+                conn->commit();
+            });
+            
+            // Make the best attempt to get rid of the temporary file. Ignore any errors
+            // for now. Just report them.
```

Comment:
Do all temporary files follow a standard naming pattern so it's easy to clean them up?